viewer/src/tools/AxisView/types.ts

```diff
@@ -0,0 +1,105 @@
+/*!
+ * Copyright 2021 Cognite AS
+ */
+
+import * as THREE from 'three';
+
+export type AxisBoxConfig = {
+  size?: number;
+  position?: Absolute | Relative;
+  faces?: {
+    xPositiveFace?: AxisBoxFaceConfig;
+    xNegativeFace?: AxisBoxFaceConfig;
+    yPositiveFace?: AxisBoxFaceConfig;
+    yNegativeFace?: AxisBoxFaceConfig;
+    zPositiveFace?: AxisBoxFaceConfig;
+    zNegativeFace?: AxisBoxFaceConfig;
+  };
+  compass?: AxisBoxCompassConfig;
+};
+
+export type Absolute = {
+  xAbsolute: number;
+  yAbsolute: number;
+};
+
+export type Relative = {
+  corner: Corner;
+  padding: THREE.Vector2;
+};
+
+export type AxisBoxFaceConfig = {
+  label?: string;
+  fontSize?: number;
+  fontColor?: THREE.Color;
+  outlineSize?: number;
+  outlineColor?: THREE.Color;
+  faceColor?: THREE.Color;
+};
+
+export type AxisBoxCompassConfig = {
+  ringLabel?: string;
+  labelDelta?: number;
+  fontSize?: number;
+  tickColor?: THREE.Color;
+  fontColor?: THREE.Color;
+};
+
+export enum Corner {
+  TopRight,
+  TopLeft,
+  BottomLeft,
+  BottomRight
+}
+
+export const defaultAxisBoxCompassConfig: AxisBoxCompassConfig = {
```

Comment:
```suggestion
export const defaultAxisBoxCompassConfig: Required<AxisBoxCompassConfig> = {
```

Reply:
This actually has some fields that intentionally could be undefined (f.ex font size will be automatically estimated based on size of component if not set).